Projects/UOContent/Mobiles/AI/BaseAI.cs

```diff
@@ -1992,100 +1988,103 @@ public virtual void WalkRandom(int iChanceToNotMove, int iChanceToDir, int iStep
+
+        /*
+         * Movement Speed Table based on RunUO
+         * <0.075 -> 0.0375
+         * 0.075 -> 0.05
+         * 0.1 -> 0.125
+         * 0.2 -> 0.3
+         * 0.25 -> 0.45
+         * 0.3 -> 0.6
+         * 0.4 -> 0.85
+         * 0.5 -> 1.05
+         * 0.6 -> 1.2,
+         * 0.8 -> 1.5
+         * 1.0 -> 1.8
+         * Above 1.0 is linear
+         */
+
+        // Linear interpolated
+        var movementSpeed = moveSpeed switch
+        {
+            >= 1.0   => 1.8 * moveSpeed,
+            >= 0.5   => 1.05 + (moveSpeed - 0.5) * 1.5,
+            >= 0.4   => 0.85 + (moveSpeed - 0.4) * 2,
+            >= 0.3   => 0.6 + (moveSpeed - 0.3) * 2.5,
+            >= 0.2   => 0.3 + (moveSpeed - 0.2) * 3,
+            >= 0.1   => 0.125 + (moveSpeed - 0.1) * 1.75,
+            >= 0.075 => 0.05 + (moveSpeed - 0.075) * 3,
+            _        => 0.0375 // 30 ticks, 37.5ms
+        };
 
-        // Monster is passive
-        if (!isControlled && Math.Abs(thinkingSpeed - m_Mobile.PassiveSpeed) < 0.0001)
+        if (isPassive)
         {
-            thinkingSpeed *= 3; // Monster passive is 3x slower than thinking
+            movementSpeed += 0.2;
         }
-        else if (!isControlled || m_Mobile.ControlOrder != OrderType.Follow || m_Mobile.ControlTarget != m_Mobile.ControlMaster)
+
+        if (!isControlled)
+        {
+            movementSpeed += 0.1;
+        }
+        else if (!bc.Summoned)
         {
-            thinkingSpeed *= 2; // Monster active speed is 2x slower than thinking
+            if (bc.ControlOrder == OrderType.Follow && bc.ControlTarget == bc.ControlMaster)
+            {
+                movementSpeed *= 0.5;
+            }
+
+            movementSpeed -= 0.075;
         }
 
-        if (!m_Mobile.IsDeadPet && (m_Mobile.ReduceSpeedWithDamage || m_Mobile.IsSubdued))
+        if (!bc.IsDeadPet && (bc.ReduceSpeedWithDamage || bc.IsSubdued))
         {
             int stats, statsMax;
             if (Core.HS)
             {
-                stats = m_Mobile.Stam;
-                statsMax = m_Mobile.StamMax;
+                stats = bc.Stam;
+                statsMax = bc.StamMax;
             }
             else
             {
-                stats = m_Mobile.Hits;
-                statsMax = m_Mobile.HitsMax;
+                stats = bc.Hits;
+                statsMax = bc.HitsMax;
             }
 
             var offset = statsMax <= 0 ? 1.0 : Math.Max(0, stats) / (double)statsMax;
 
-            if (offset < 1.0)
-            {
-                thinkingSpeed += m_Mobile.PassiveSpeed * (1.0 - offset);
-            }
+            movementSpeed += (1.0 - offset) * 0.8;
         }
 
-        return thinkingSpeed;
+        return movementSpeed;
     }
 
-    public virtual bool CheckMove() => Core.TickCount - NextMove >= 0;
+    private long TimeUntilNextMove() => Math.Max(NextMove - Core.TickCount, 0);
+
+    public virtual bool CheckMove() => true;
```

Comment:
We are _removing_ the time until check from `CheckMove` so we can use it to schedule a movement in between "thinks".